boto3/dynamodb/transform.py

```diff
@@ -0,0 +1,188 @@
+# Copyright 2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from collections import Mapping
+import functools
+
+from boto3.dynamodb.types import TypeSerializer, TypeDeserializer
+from boto3.dynamodb.conditions import ConditionBase
+from boto3.dynamodb.conditions import ConditionExpressionBuilder
+
+
+def register_high_level_interface(base_classes, **kwargs):
+    # This assumes the base class is the base resource object
+    resource_base_class = base_classes[0]
+
+    class DynamoDBHighLevelResource(resource_base_class):
```

Comment:
As I was reading this, I was curious why this wasn't just its once class at the module level?  It would avoid having to execute the same code every time a ddb resource was created, and would allow us to unit test this class directly and make the code a little cleaner IMO.  Note a huge deal but I was curious.